services/webdav/pkg/service/v0/search.go

```diff
@@ -218,6 +222,50 @@ func matchToPropResponse(ctx context.Context, match *searchmsg.Match) (*propfind
 	return &response, nil
 }
 
+func addNonEmptyIntProp[T ~int | ~int8 | ~int16 | ~int32 | ~int64](propstatOK *propfind.PropstatXML, name string, value *T) {
+	if value == nil {
+		return
+	}
+	propstatOK.Prop = append(propstatOK.Prop, prop.Raw(name, strconv.FormatInt(int64(*value), 10)))
+}
+
+func addNonEmptyStrProp(propstatOK *propfind.PropstatXML, name string, value *string) {
+	if value == nil {
+		return
+	}
+	propstatOK.Prop = append(propstatOK.Prop, prop.Escaped(name, *value))
+}
+
+func addNonEmptyBoolProp(propstatOK *propfind.PropstatXML, name string, value *bool) {
+	if value == nil {
+		return
+	}
+	propstatOK.Prop = append(propstatOK.Prop, prop.Raw(name, strconv.FormatBool(*value)))
+}
+
+func addAudioProps(propstatOK *propfind.PropstatXML, entity *searchmsg.Entity) {
+	if entity.Audio == nil {
+		return
+	}
+
+	addNonEmptyStrProp(propstatOK, "oc:audio.album", entity.Audio.Album)
+	addNonEmptyStrProp(propstatOK, "oc:audio.albumArtist", entity.Audio.AlbumArtist)
+	addNonEmptyStrProp(propstatOK, "oc:audio.artist", entity.Audio.Artist)
+	addNonEmptyIntProp(propstatOK, "oc:audio.bitrate", entity.Audio.Bitrate)
+	addNonEmptyStrProp(propstatOK, "oc:audio.composers", entity.Audio.Composers)
+	addNonEmptyStrProp(propstatOK, "oc:audio.copyright", entity.Audio.Copyright)
+	addNonEmptyIntProp(propstatOK, "oc:audio.disc", entity.Audio.Disc)
+	addNonEmptyIntProp(propstatOK, "oc:audio.discCount", entity.Audio.DiscCount)
+	addNonEmptyIntProp(propstatOK, "oc:audio.duration", entity.Audio.Duration)
+	addNonEmptyStrProp(propstatOK, "oc:audio.genre", entity.Audio.Genre)
+	addNonEmptyBoolProp(propstatOK, "oc:audio.hasDrm", entity.Audio.HasDrm)
+	addNonEmptyBoolProp(propstatOK, "oc:audio.isVariableBitrate", entity.Audio.IsVariableBitrate)
+	addNonEmptyStrProp(propstatOK, "oc:audio.title", entity.Audio.Title)
+	addNonEmptyIntProp(propstatOK, "oc:audio.track", entity.Audio.Track)
+	addNonEmptyIntProp(propstatOK, "oc:audio.trackCount", entity.Audio.TrackCount)
+	addNonEmptyIntProp(propstatOK, "oc:audio.year", entity.Audio.Year)
```

Comment:
Maybe we should not roll this out, but have something like
```
<oc:audio>
  <oc:album>...</oc:album>
</oc:audio>
```

Which would make it easier to just request the audio prop and would not require to request every single one individually...?

Reply:
Good idea. The WebDAV Spec seems to support that pattern.

Reply:
Cool! Would like to delay this to a follow up anyhow